Build error

maybe needs -fPIC?

```
Consolidate compiler generated dependencies of target mts-status
[ 25%] Linking CXX executable mts-status
/usr/bin/ld: CMakeFiles/mts-status.dir/libs/MTS-ESP/Client/libMTSClient.cpp.o: in function `mtsclientglobal::load_lib()':
libMTSClient.cpp:(.text._ZN15mtsclientglobal8load_libEv[_ZN15mtsclientglobal8load_libEv]+0x19): undefined reference to `dlopen'
/usr/bin/ld: libMTSClient.cpp:(.text._ZN15mtsclientglobal8load_libEv[_ZN15mtsclientglobal8load_libEv]+0x45): undefined reference to `dlopen'
/usr/bin/ld: libMTSClient.cpp:(.text._ZN15mtsclientglobal8load_libEv[_ZN15mtsclientglobal8load_libEv]+0x8e): undefined reference to `dlsym'
/usr/bin/ld: libMTSClient.cpp:(.text._ZN15mtsclientglobal8load_libEv[_ZN15mtsclientglobal8load_libEv]+0xb3): undefined reference to `dlsym'
/usr/bin/ld: libMTSClient.cpp:(.text._ZN15mtsclientglobal8load_libEv[_ZN15mtsclientglobal8load_libEv]+0xd8): undefined reference to `dlsym'
/usr/bin/ld: libMTSClient.cpp:(.text._ZN15mtsclientglobal8load_libEv[_ZN15mtsclientglobal8load_libEv]+0xfd): undefined reference to `dlsym'
/usr/bin/ld: libMTSClient.cpp:(.text._ZN15mtsclientglobal8load_libEv[_ZN15mtsclientglobal8load_libEv]+0x122): undefined reference to `dlsym'
/usr/bin/ld: CMakeFiles/mts-status.dir/libs/MTS-ESP/Client/libMTSClient.cpp.o:libMTSClient.cpp:(.text._ZN15mtsclientglobal8load_libEv[_ZN15mtsclientglobal8load_libEv]+0x147): more undefined references to `dlsym' follow
/usr/bin/ld: CMakeFiles/mts-status.dir/libs/MTS-ESP/Client/libMTSClient.cpp.o: in function `mtsclientglobal::~mtsclientglobal()':
libMTSClient.cpp:(.text._ZN15mtsclientglobalD2Ev[_ZN15mtsclientglobalD5Ev]+0x39): undefined reference to `dlclose'
/usr/bin/ld: CMakeFiles/mts-status.dir/libs/MTS-ESP/Master/libMTSMaster.cpp.o: in function `mtsmasterglobal::load_lib()':
libMTSMaster.cpp:(.text._ZN15mtsmasterglobal8load_libEv[_ZN15mtsmasterglobal8load_libEv]+0x19): undefined reference to `dlopen'
/usr/bin/ld: libMTSMaster.cpp:(.text._ZN15mtsmasterglobal8load_libEv[_ZN15mtsmasterglobal8load_libEv]+0x45): undefined reference to `dlopen'
/usr/bin/ld: libMTSMaster.cpp:(.text._ZN15mtsmasterglobal8load_libEv[_ZN15mtsmasterglobal8load_libEv]+0x8e): undefined reference to `dlsym'
/usr/bin/ld: libMTSMaster.cpp:(.text._ZN15mtsmasterglobal8load_libEv[_ZN15mtsmasterglobal8load_libEv]+0xb2): undefined reference to `dlsym'
/usr/bin/ld: libMTSMaster.cpp:(.text._ZN15mtsmasterglobal8load_libEv[_ZN15mtsmasterglobal8load_libEv]+0xd7): undefined reference to `dlsym'
/usr/bin/ld: libMTSMaster.cpp:(.text._ZN15mtsmasterglobal8load_libEv[_ZN15mtsmasterglobal8load_libEv]+0xfc): undefined reference to `dlsym'
/usr/bin/ld: libMTSMaster.cpp:(.text._ZN15mtsmasterglobal8load_libEv[_ZN15mtsmasterglobal8load_libEv]+0x121): undefined reference to `dlsym'
/usr/bin/ld: CMakeFiles/mts-status.dir/libs/MTS-ESP/Master/libMTSMaster.cpp.o:libMTSMaster.cpp:(.text._ZN15mtsmasterglobal8load_libEv[_ZN15mtsmasterglobal8load_libEv]+0x146): more undefined references to `dlsym' follow
/usr/bin/ld: CMakeFiles/mts-status.dir/libs/MTS-ESP/Master/libMTSMaster.cpp.o: in function `mtsmasterglobal::~mtsmasterglobal()':
libMTSMaster.cpp:(.text._ZN15mtsmasterglobalD2Ev[_ZN15mtsmasterglobalD5Ev]+0x2b): undefined reference to `dlclose'
collect2: error: ld returned 1 exit status
gmake[2]: *** [CMakeFiles/mts-status.dir/build.make:129: mts-status] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:83: CMakeFiles/mts-status.dir/all] Error 2
gmake: *** [Makefile:91: all] Error 2

```

--- a/.github/workflows/build-cmake.yml
+++ b/.github/workflows/build-cmake.yml
@@ -1,0 +1,36 @@
+name: CI-Elastika-CMak
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+  workflow_dispatch:
+
+jobs:
+  build_and_test:
+    name: Build Elastika
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false # show all errors for each platform (vs. cancel jobs on error)
+      matrix:
+        os: [ubuntu-latest, windows-latest, macOS-latest]
+
+    steps:
+      - name: Get latest CMake
+        uses: lukka/get-cmake@latest
+
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          submodules: recursive
+
+      - name: Configure
+        shell: bash
+        run: cmake -Bbuild -DCMAKE_BUILD_TYPE=Release 
+
+      - name: Build
+        shell: bash
+        run: cmake --build build --config Release --parallel 4 

--- a/.github/workflows/build-cmake.yml
+++ b/.github/workflows/build-cmake.yml
@@ -1,4 +1,4 @@
-name: CI-Elastika-CMak
+name: CI-Cmake-MTS-Status
 
 on:
   push:

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -8,4 +8,8 @@ add_executable(${PROJECT_NAME}
                libs/MTS-ESP/Client/libMTSClient.cpp
                libs/MTS-ESP/Master/libMTSMaster.cpp)
 target_include_directories(${PROJECT_NAME} PUBLIC libs/MTS-ESP/Client libs/MTS-ESP/Master)
+
+if (UNIX AND NOT APPLE)
+   target_link_libraries(${PROJECT_NAME} PUBLIC dl)
+endif()
                

